src/main/java/io/gravitee/policy/jsonvalidation/JsonValidationPolicy.java

```diff
@@ -42,6 +46,10 @@
 
     static final String JSON_INVALID_PAYLOAD_KEY = "JSON_INVALID_PAYLOAD";
     static final String JSON_INVALID_FORMAT_KEY = "JSON_INVALID_FORMAT";
+    static final String JSON_INVALID_RESPONSE_PAYLOAD_KEY = "JSON_INVALID_RESPONSE_PAYLOAD";
```

Comment:
Do we have to manage distincts MESSAGE_KEY depending on the scope ?
Why not preserving a single KEY per scope (request / response) and provide additional parameters to specify the scope?
I mean, something like this: https://github.com/gravitee-io/gravitee-policy-ratelimit/blob/master/gravitee-policy-ratelimit/src/main/java/io/gravitee/policy/ratelimit/RateLimitPolicy.java#L187

WDYT?

Reply:
But providing single key will prevent from overriding response (http code mainly) for different scopes in different ways through response templates mechanism. Am I wrong?

Reply:
I think I understand your purpose.
So yes, if we are using a single key, the same response template will be used for both on-request and on-response scope, you are right